spec/support/acceptance_test_helpers.rb

```diff
@@ -43,6 +50,15 @@ module AcceptanceTestHelpers
     end
   end
 
+  # A head version of MRI ruby will require the -dev appended to the ruby string in the gemfile.
+  # Unfortunately, there is no ENV variable that has access to the "dev" concept directly.
+  # Perhaps the best we can do is parse RUBY_DESCRIPTION?
+  #   > RUBY_DESCRIPTION
+  #   "ruby 3.5.0dev (2025-02-20T18:14:37Z ... etc)
+  def ruby_dev_append
+    RUBY_DESCRIPTION.include?("dev") ? "-dev" : ""
+  end
```

Comment:
The `head` build wouldn't pass without this hack.  The specs that test seetting the Ruby version directly in the Gemfile need that line to say `3.5.0-dev`, or Bundler will think that `3.5.0` and `3.5.0-dev` are different versions of Ruby, and exit prematurely.